controllers/managedfusion_controller.go

```diff
@@ -432,6 +444,11 @@ func (r *ManagedFusionReconciler) reconcilePrometheus() error {
 
 		r.prometheus.Spec = desired.Spec
 		r.prometheus.Spec.ExternalLabels["clusterId"] = string(clusterVersion.Spec.ClusterID)
+		if r.remoteWriteConfig.BearerToken != "" {
+			r.prometheus.Spec.RemoteWrite[0].BearerToken = r.remoteWriteConfig.BearerToken
+		} else {
+			r.prometheus.Spec.RemoteWrite[0] = promv1.RemoteWriteSpec{}
+		}
```

Comment:
I believe the check here is wrong. 
This will disable remote write even in the case where there is a remote write config in the secret but the bearer token is missing. This will be considered a bug. 

You need to check for the existence of the section to decide if to enable or disable remote write. And as I have suggested in the couple of my previous comments that can be done by:
```suggestion
		if r.remoteWriteConfig != nil {
			if .r.remoteWriteConfig.BearerToken == "" {
				return fmt.Errorf("Invalid bearer token at remote write config")
			}
			r.prometheus.Spec.RemoteWrite[0].BearerToken = r.remoteWriteConfig.BearerToken
		} else {
			r.prometheus.Spec.RemoteWrite[0] = promv1.RemoteWriteSpec{}
		}
```

Reply:
Made the change as per suggestion